src/libmongoc/src/mongoc/mongoc-cluster-private.h

```diff
@@ -43,14 +43,14 @@ BSON_BEGIN_DECLS
 typedef struct _mongoc_cluster_node_t {
    mongoc_stream_t *stream;
    char *connection_address;
+   uint32_t generation;
```

Comment:
Would it make sense to name this `pool_generation` or is `mongoc_cluster_node_t` synonymous with a pool?

Perhaps this makes more sense for the `generation` field on the SD object, as I was getting confused as to whether that corresponded to a pool or not. Ideally, should an SD always have the same `generation` as its cluster unless it's stale?

Reply:
Thanks for the explanation. In this case I think the generic `generation` name would be preferable, as I was definitely misunderstanding the relationship between a cluster and pool.

Reply:
Gotcha, renamed it back to `generation`